starcheck/src/lib/Ska/Starcheck/Obsid.pm

```diff
@@ -2861,7 +2883,28 @@ sub quat2radecroll {
 sub check_guide_count {
 ###################################################################################
     my $self = shift;
-    my $guide_count = $self->count_guide_stars();
+
+    # Disable dynamic background if guide dither is disabled or amplitude is small
+    my $guide_dither = $self->{dither_guide};
+    my $dyn_bgd;
+    my $guide_dither_ampl_y = int($guide_dither->{ampl_y} + 0.5);
+    my $guide_dither_ampl_p = int($guide_dither->{ampl_p} + 0.5);
+    if (   ($guide_dither->{state} eq 'DISA')
+        or (($guide_dither_ampl_y == 0) and ($guide_dither_ampl_p == 0)))
```

Comment:
I thought at some point we were going to set dyn_bgd=0 for dither < 8 in both pitch ,yaw. E.g. if dither is (4, 4) then dyn bgd is not effective. We recognize that this is stricter than MATLAB tools / sparkles, but what we want see for guide count in starcheck is the more "truthful" evaluation without the dyn bgd bonus. This saves us from doing the calculation by hand, which is what my action would be from the warning issued by the earlier code.

Reply:
We went back and forth on this.  It sounds like what I heard as the final strategy (compute guide count with dynamic background off for zero dither but give at least some kind of warning for the completely unexpected state of small dither) isn't quite what you heard - which is fine.  The middle ground of 4x4 dither should just not happen at all so also fine to update this calculation for the < 8 regime.  Sparkles is just set for the == 0 (rounded) case in https://github.com/sot/sparkles/pull/221 so that could also get changed or left alone.  Either way.

Reply:
Ok, I set the starcheck count to just use the < 8 check here.